include/CL/sycl/detail/small_array.hpp

```diff
@@ -35,7 +35,12 @@ namespace detail {
     for (std::size_t i = 0; i != Dims; ++i)             \
       (*this)[i] op rhs[i];                             \
     return *this;                                       \
-  }
+  }                                                     \
+  FinalType operator op(const size_t &rhs) {            \
```

Comment:
I have the feeling that what we want for genericity is actually `BasicType` instead of all these `size_t` everywhere.

Reply:
The SYCL spec is quite clear on size_t, but yeah I suppose for small_array genericity you might be right or I just move the size_t operators to the id/range subclasses?